ceph/rados/monitor_workflows.py

```diff
@@ -281,3 +282,107 @@ def set_tiebreaker_mon(self, host) -> bool:
             return False
         log.info("Successfully set the new tiebreaker mon on cluster")
         return True
+
+    def connection_score_checks(
+        self, hosts, rados_obj, num_mons, mon_election_obj
+    ) -> bool:
+        """
+        Checks all mon daemons for correct number of monitors, correct number of peer monitors,
+        correct peer monitors exist for every monitor and valid connection scores for every
+        monitor ( connection scores in the range of 0 to 1 )
+
+        Example:
+            rados_obj = RadosOrchestrator(node=cephadm)
+            mon_election_obj = MonElectionStrategies(rados_obj=rados_obj)
+            mon_nodes = ceph_cluster.get_nodes(role="mon")
+            mon_nums = len(mon_nodes)
+
+            if not mon_workflow_obj.connection_score_checks(
+                mon_nodes, rados_obj, mon_nums, mon_election_obj
+            ):
+                raise Exception("Connection score checks failed")
+
+        Args:
+            hosts: Cluster object of all monitor host
+            rados_obj: RadosOrchestrator object
+            num_mons: Number of mons to exist in the cluster
+            mon_election_obj: MonElectionStrategies object
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Collect mon host and rank from ceph mon dump command
+        mon_quorum = mon_election_obj.get_mon_quorum()
+
+        for host in hosts:
+
+            if not rados_obj.check_daemon_exists_on_host(
+                host=host.hostname, daemon_type="mon"
+            ):
+                continue
+
+            mon_status, mon_status_desc = rados_obj.get_daemon_status(
+                daemon_type="mon", daemon_id=host.hostname
+            )
+
+            if mon_status != 1 or mon_status_desc != "running":
+                continue
+
+            cmd = (
+                f"cephadm shell ceph daemon mon.{host.hostname} connection scores dump"
+            )
+            out, err = host.exec_command(sudo=True, cmd=cmd)
+            mon_conn_score = json.loads(out)
```

Comment:
Maybe add a try - catch block here. 

In case the command execution failed, we will see `JSONDecodeError` here. Let's plan and avoid command execution errors.